File: framework/src/main/java/com/github/mrmitew/bodylog/repository/di/RepositoryModule.java
package com.github.mrmitew.bodylog.repository.di;

import com.github.mrmitew.bodylog.data.repository.ProfileRepository;
import com.github.mrmitew.bodylog.domain.repository.Repository;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class RepositoryModule {
    @Provides
    @Singleton
    Repository providesRepository() {
        return new ProfileRepository();
    }
}

File: framework/src/main/java/com/github/mrmitew/bodylog/framework/repository/di/RepositoryModule.kt
package com.github.mrmitew.bodylog.framework.repository.di

import com.github.mrmitew.bodylog.data.repository.ProfileRepository
import com.github.mrmitew.bodylog.domain.repository.Repository
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class RepositoryModule {
    @Provides
    @Singleton internal fun providesRepository(): Repository = ProfileRepository()
}